### PR TITLE
spider: add spider for São Gonçalo, RJ

### PR DESCRIPTION
**Pull request by chicocvenancio**

### CODE REVIEW

File: processing/data_collection/gazette/spiders/rj_sao_goncalo.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    allowed_domains = ['www.pmsg.rj.gov.br']
+    download_gazette_url = 'http://www.pmsg.rj.gov.br/diario/{}.pdf'
+    start_urls = ['http://www.pmsg.rj.gov.br/diario_oficial.php']
+
+    def parse(self, response):
+        date = datetime.date.today()
+        end_date = datetime.date(2015, 1, 1)
+        while date >= end_date:
+            self.logger.info("got gazette for day %s", date)
+            url = self.download_gazette_url.format(date.strftime('%Y_%m_%d'))
```
Comment (giovanisleite) [inline processing/data_collection/gazette/spiders/rj_sao_goncalo.py]: url pattern :+1:

File: processing/data_collection/gazette/spiders/rj_sao_goncalo.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class RjSaoGoncaloSpider(BaseGazetteSpider):
+    TERRITORY_ID = '3304904'
+    name = 'rj_sao_goncalo'
+    allowed_domains = ['www.pmsg.rj.gov.br']
+    download_gazette_url = 'http://www.pmsg.rj.gov.br/diario/{}.pdf'
+    start_urls = ['http://www.pmsg.rj.gov.br/diario_oficial.php']
+
+    def parse(self, response):
+        date = datetime.date.today()
+        end_date = datetime.date(2015, 1, 1)
```
Comment (giovanisleite) [inline processing/data_collection/gazette/spiders/rj_sao_goncalo.py]: Change this date to the first document available date (`03/02/1998`, I think) and let the DropItem (In the pipeline) take care of the period of interest (That can be mutable so let the pipeline controls it helps with the maintenance)

File: processing/data_collection/gazette/spiders/rj_sao_goncalo.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+from gazette.items import Gazette
+import datetime
+from gazette.spiders.base import BaseGazetteSpider
+
```
Comment (giovanisleite) [inline processing/data_collection/gazette/spiders/rj_sao_goncalo.py]: We can follow [PEP8 guideline for imports](https://www.python.org/dev/peps/pep-0008/#imports) here, so:
```
import datetime

from gazette.spiders.base import BaseGazetteSpider
from gazette.items import Gazette
```

File: processing/data_collection/gazette/spiders/rj_sao_goncalo.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        while date >= end_date:
+            self.logger.info("got gazette for day %s", date)
+            url = self.download_gazette_url.format(date.strftime('%Y_%m_%d'))
+            is_extra_edition = False
+            yield Gazette(
+                date=date,
+                file_urls=[url],
+                is_extra_edition=is_extra_edition,
+                territory_id=self.TERRITORY_ID,
+                power='executive',
```
Comment (giovanisleite) [inline processing/data_collection/gazette/spiders/rj_sao_goncalo.py]: I have not found any example with legislature actions, so power confirmed

File: processing/data_collection/gazette/spiders/rj_sao_goncalo.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        date = datetime.date.today()
+        end_date = datetime.date(2015, 1, 1)
+        while date >= end_date:
+            self.logger.info("got gazette for day %s", date)
+            url = self.download_gazette_url.format(date.strftime('%Y_%m_%d'))
+            is_extra_edition = False
+            yield Gazette(
+                date=date,
+                file_urls=[url],
+                is_extra_edition=is_extra_edition,
```
Comment (giovanisleite) [inline processing/data_collection/gazette/spiders/rj_sao_goncalo.py]: Apparently there is no control about it, so :+1:

**Comment by chicocvenancio** [comment]
@giovanisleite thanks for the comments, fixed the PEP8 imports and the end_date.

**Comment by vitorbaptista** [comment]
I updated the code with the latest main branch, and fixed the spider (their website changed). Thanks for the great work, @chicocvenancio ! I'm closing this now, as all of your code is in the new PR at https://github.com/okfn-brasil/querido-diario/pull/292

**Pull request closed without merging**